File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def gcd(a, b):
    while b:
        a, b = b, a % b
    return a

def lcm(a, b):
    return (a * b) // gcd(a, b)

def solution(n, m):
    a, b = [n, m] if n > m else [m, n]
    return [gcd(a, b), lcm(a, b)]

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int a, b;
        if (n > m) {
            a = n; b = m;
        } else {
            a = m; b = n;
        }
        
        int[] answer = {gcd(a, b), lcm(a, b)};
        return answer;
    }
    
    public int gcd(int a, int b) {
        int temp;
        while (b != 0) {
            temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
    
    public int lcm(int a, int b) {
        return (a * b) / gcd(a, b);
    }
}